shipper.go

```diff
@@ -14,11 +14,220 @@ import (
 	minio "github.com/minio/minio-go"
 	"github.com/pkg/errors"
 	"github.com/prometheus/tsdb"
+	"gopkg.in/fsnotify.v1"
 )
 
 var prefix = "promblock-"
 
-func startShipper(configFile string, logger log.Logger) {
+func refresh(dirname string, logger log.Logger) []string {
+
+	blockDirs, err := ioutil.ReadDir(dirname)
+	if err != nil {
+		level.Error(logger).Log("error", err.Error())
+		os.Exit(1)
+	}
+
+	blocks := make([]string, 0, len(blockDirs))
+	for _, bd := range blockDirs {
+
+		if !isBlockDir(bd) {
+			continue
+		}
+
+		byt, err := ioutil.ReadFile(filepath.Join(dirname, bd.Name(), "meta.json"))
+		if err != nil {
+			level.Error(logger).Log("msg", "reload tempdir", "error", err.Error())
+			os.Exit(1)
+		}
+
+		var bm tsdb.BlockMeta
+		if err := json.Unmarshal(byt, &bm); err != nil {
+			level.Error(logger).Log("msg", "reload tempdir", "error", err.Error())
+			os.Exit(1)
+		}
+
+		if bm.Compaction.Level == 1 {
+			blocks = append(blocks, filepath.Join(dirname, bd.Name()))
+		}
+	}
+	return blocks
+}
+
+func lazyShip(dataDir string, lazyDataDir string, logger log.Logger, done <-chan bool) {
+
+	watcher, err := fsnotify.NewWatcher()
+	defer watcher.Close()
+	if err != nil {
+		level.Error(logger).Log("msg", "lazy Ship", "error", err.Error())
+	}
+	watcher.Add(dataDir)
+
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case event := <-watcher.Events:
+
+			if event.Op&fsnotify.Create == fsnotify.Create {
```

Comment:
Can we just `reload()` the whole dir on an event instead of acting on that single dir? Its much simpler and has very little side-effect AFAICS.

Reply:
dataDir is already a parameter! I felt reloading the whole directory every time will be unnecessary overload, that's why preferred it this way. I don't see any side-effect this way as anyways we will be checking the whole directory after every particular interval of time.